src/EmailCampaigns/NewPublishPost/NewPublishPost.php

```diff
@@ -112,6 +112,13 @@ public function new_publish_post($new_status, $old_status, $post)
                 $email_campaign_id = absint($npp_campaign['id']);
 
                 if (ER::is_campaign_active($email_campaign_id) === false) continue;
+    
+                //post translations
+                $post = apply_filters('mailoptin_email_campaign_translated_post',
```

Comment:
@ShoboySnr I see you are using this filter. Can you explain on a higher level how you are using it so I can better rename this filter?

Reply:
@w3guy what it does is to find all related translations of that post and then picks the language that was selected in the customizer (if found) and then use the post instead https://github.com/mailoptin/libsodium/pull/13/files#r772105671

Reply:
@ShoboySnr in that case, remove `ER::get_merged_customizer_value($email_campaign_id, 'post_translation')` from the filter. instead use it in the filter callback in WPML.php to get the post translation.

Remember, all wpml logic must not be in Core.

Reply:
@w3guy Okay, will do.